msteams-platform/toolkit/TeamsFx-SDK.md

```diff
@@ -188,183 +235,76 @@ try {
-const config = await getTediousConnectionConfig(teamsfx);
-// If there are multiple SQL databases
-const config2 = await getTediousConnectionConfig(teamsfx "your database name");
-const connection = new Connection(config);
-connection.on("connect", (error) => {
-  if (error) {
-    console.log(error);
-  }
-});
-```
-
-</details>
-
-<br>
-
-<details>
-<summary><b>Use certificate-based authentication in Azure Function</b></summary>
-
-```ts
-const authConfig = {
-  clientId: process.env.M365_CLIENT_ID,
-  certificateContent: "The content of a PEM-encoded public/private key certificate",
-  authorityHost: process.env.M365_AUTHORITY_HOST,
-  tenantId: process.env.M365_TENANT_ID,
-};
-const teamsfx = new TeamsFx(IdentityType.App);
-teamsfx.setCustomeConfig({
-  certificateContent: "The content of a PEM-encoded public/private key certificate"
-});
-const token = teamsfx.getCredential().getToken();
-```
-
-</details>
-
-<br>
-
-<details>
-<summary><b>Use Graph API in bot application</b></summary>
+    ```ts
+    import {
+      createMicrosoftGraphClient,
+      TeamsFx,
+    } from "@microsoft/teamsfx";
+    ```
 
-Add `TeamsBotSsoPrompt` to dialog set.
+    3. Use `TeamsFx.login()` to get user consent.
 
-```ts
-const { ConversationState, MemoryStorage } = require("botbuilder");
-const { DialogSet, WaterfallDialog } = require("botbuilder-dialogs");
-const { TeamsBotSsoPrompt } = require("@microsoft/teamsfx");
+    ```ts
+    // Put these code in a call-to-action callback function to avoid browser blocking automatically showing up pop-ups.
+    await teamsfx.login(["User.Read"]); // Login with scope
+    ```
 
-const convoState = new ConversationState(new MemoryStorage());
-const dialogState = convoState.createProperty("dialogState");
-const dialogs = new DialogSet(dialogState);
+    3. Initial a TeamsFx instance and graph client and get information from MS Graph by this client.
 
-const teamsfx = new TeamsFx();
-dialogs.add(
-  new TeamsBotSsoPrompt(teamsfx, "TeamsBotSsoPrompt", {
-    scopes: ["User.Read"],
-  })
-);
-
-dialogs.add(
-  new WaterfallDialog("taskNeedingLogin", [
-    async (step) => {
-      return await step.beginDialog("TeamsBotSsoPrompt");
-    },
-    async (step) => {
-      const token = step.result;
-      if (token) {
-        // ... continue with task needing access token ...
-      } else {
-        await step.context.sendActivity(`Sorry... We couldn't log you in. Try again later.`);
-        return await step.endDialog();
+    ```ts
+    try {
+      const teamsfx = new TeamsFx();
+      const graphClient = createMicrosoftGraphClient(teamsfx, ["User.Read"]); // Initializes MS Graph SDK using our MsGraphAuthProvider
+      const profile = await graphClient.api("/me").get();
+    } catch (err: unknown) {
+      // ErrorWithCode is handled by Graph client
+      if (err instanceof GraphError && err.code?.includes(ErrorCode.UiRequiredError)) {
+        this.setState({
+          showLoginBtn: true,
+        });
```

Comment:
this.setState is the code related to react, you can remove this and just add comments that: "show login button to ask for user consent"